src/MIDebugEngine/Engine.Impl/ExceptionManager.cs

```diff
@@ -480,17 +495,32 @@ private async Task UpdateCatagory(Guid categoryId, ExceptionCategorySettings cat
                     }
                 }
 
-                IEnumerable<ulong> breakpointIds = await _commandFactory.SetExceptionBreakpoints(categoryId, exceptionNames, grouping.Key);
-
-                int count = exceptionNames.Zip(breakpointIds, (exceptionName, breakpointId) =>
+                if (grouping.Key.ToString().Contains(ExceptionBreakpointStates.BreakThrown.ToString())) // only set catchpoints on BreakThrown - problematic when grouping.Key is BreakThrown | something
```

Comment:
>grouping.Key.ToString().Contains(ExceptionBreakpointStates.BreakThrown.ToString()) [](http://example.com/codeflow?start=20&length=82)

You don't want to turn things into a string to see if it contains a flag.

`grouping.Key.HasFlag(ExceptionBreakpointStates.BreakThrown)`

Though really we can get rid of this whole `.GroupBy` code now since we only care about rules with `grouping.Key.HasFlag(ExceptionBreakpointStates.BreakThrown)` except to remember that we are in an unsupported state so we need to warn when we stop.